python/ray/experimental/data/dataset.py

```diff
@@ -195,6 +196,117 @@ def repartition(self, num_blocks: int) -> "Dataset[T]":
         new_blocks = simple_shuffle(self._blocks, num_blocks)
         return Dataset(new_blocks)
 
+    def split(self, n: int,
+              locality_hints: List[Any] = None) -> List["Dataset[T]"]:
+        """Split the dataset into ``n`` disjoint pieces.
+
+        This returns a list of sub-datasets that can be passed to Ray tasks
+        and actors and used to read the dataset records in parallel.
+
+        Examples:
+            >>> # Split up a dataset to process over `n` worker actors.
+            >>> shards = ds.split(len(workers), locality_hints=workers)
+            >>> for shard, worker in zip(shards, workers):
+            ...     worker.consume.remote(shard)
+
+        Time complexity: O(1)
+
+        Args:
+            n: Number of child datasets to return.
+            locality_hints: A list of Ray actor handles of size ``n``. The
+                system will try to co-locate the blocks of the ith dataset
+                with the ith actor to maximize data locality.
+
+        Returns:
+            A list of ``n`` disjoint dataset splits.
+        """
+        assert n > 0 and n <= len(self._blocks)
+        assert locality_hints is None or len(locality_hints) == n
```

Comment:
raise ValueError("The length of `locality_hints` must equal the number of splits.")